modules/java/proxies/java-destination-proxy.c

```diff
@@ -192,14 +192,15 @@ java_destination_proxy_free(JavaDestinationProxy *self)
 }
 
 JavaDestinationProxy *
-java_destination_proxy_new(const gchar *class_name, const gchar *class_path, gpointer handle, LogTemplate *template)
+java_destination_proxy_new(const gchar *class_name, const gchar *class_path, gpointer handle, LogTemplate *template,
+                           const gchar *jvm_options)
 {
   JavaDestinationProxy *self = g_new0(JavaDestinationProxy, 1);
   self->java_machine = java_machine_ref();
   self->formatted_message = g_string_sized_new(1024);
   self->template = log_template_ref(template);
 
-  if (!java_machine_start(self->java_machine))
+  if (!java_machine_start(self->java_machine, jvm_options))
```

Comment:
`self->java_machine` points to a "singleton" object, so only the last (or first?) `jvm-options()` usage in the configuration will be in effect.

Shouldn't we use a global option instead?
We can keep this per-destination option as well, but I would not document this in the admin guide due to its misleading behavior.